core/kernel/notif_default.c

```diff
@@ -1,97 +1,169 @@
+	if (!nvb) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto out;
+	}
+
 	old_itr_status = cpu_spin_lock_xsave(&notif_default_lock);
 
-	if (!alloc_values_inited) {
-		bit_set(notif_alloc_values, NOTIF_VALUE_DO_BOTTOM_HALF);
-		alloc_values_inited = true;
+	if (!nvb->alloc_values_inited) {
+		bit_set(nvb->alloc_values, NOTIF_VALUE_DO_BOTTOM_HALF);
+		nvb->alloc_values_inited = true;
 	}
 
-	bit_ffc(notif_alloc_values, (int)NOTIF_ASYNC_VALUE_MAX + 1, &bit);
-	if (bit >= 0) {
-		*val = bit;
-		bit_set(notif_alloc_values, bit);
+	bit_ffc(nvb->alloc_values, (int)NOTIF_ASYNC_VALUE_MAX + 1, &bit);
+	if (bit < 0) {
+		res = TEE_ERROR_OUT_OF_MEMORY;
+		goto out_unlock;
 	}
+	*val = bit;
+	bit_set(nvb->alloc_values, bit);
 
+out_unlock:
 	cpu_spin_unlock_xrestore(&notif_default_lock, old_itr_status);
+out:
+	virt_put_guest(prtn);
 
-	if (bit < 0)
-		return TEE_ERROR_OUT_OF_MEMORY;
-
-	return TEE_SUCCESS;
+	return res;
 }
 
 void notif_free_async_value(uint32_t val)
 {
+	struct guest_partition *prtn = NULL;
+	struct notif_vm_bitmap *nvb = NULL;
 	uint32_t old_itr_status = 0;
 
+	prtn = virt_get_current_guest();
+	nvb = get_notif_vm_bitmap(prtn);
+	if (!nvb)
+		goto out;
+
 	old_itr_status = cpu_spin_lock_xsave(&notif_default_lock);
 
 	assert(val < NOTIF_ASYNC_VALUE_MAX);
-	assert(bit_test(notif_alloc_values, val));
-	bit_clear(notif_alloc_values, val);
+	assert(bit_test(nvb->alloc_values, val));
+	bit_clear(nvb->alloc_values, val);
 
 	cpu_spin_unlock_xrestore(&notif_default_lock, old_itr_status);
+out:
+	virt_put_guest(prtn);
 }
 
 uint32_t notif_get_value(bool *value_valid, bool *value_pending)
 {
+	struct guest_partition *prtn = NULL;
+	struct notif_vm_bitmap *nvb = NULL;
 	uint32_t old_itr_status = 0;
 	uint32_t res = 0;
-	int bit = 0;
+	int bit = -1;
+
+	prtn = virt_get_current_guest();
+	nvb = get_notif_vm_bitmap(prtn);
+	if (!nvb) {
+		*value_valid = false;
+		goto out;
+	}
 
 	old_itr_status = cpu_spin_lock_xsave(&notif_default_lock);
 
-	bit_ffs(notif_values, (int)NOTIF_ASYNC_VALUE_MAX + 1, &bit);
+	bit_ffs(nvb->values, (int)NOTIF_ASYNC_VALUE_MAX + 1, &bit);
 	*value_valid = (bit >= 0);
-	if (!*value_valid) {
-		*value_pending = false;
-		goto out;
-	}
+	if (!*value_valid)
+		goto out_unlock;
 
 	res = bit;
-	bit_clear(notif_values, res);
-	bit_ffs(notif_values, (int)NOTIF_ASYNC_VALUE_MAX + 1, &bit);
-	*value_pending = (bit >= 0);
-out:
+	bit_clear(nvb->values, res);
+	bit_ffs(nvb->values, (int)NOTIF_ASYNC_VALUE_MAX + 1, &bit);
+
+out_unlock:
 	cpu_spin_unlock_xrestore(&notif_default_lock, old_itr_status);
+out:
+	virt_put_guest(prtn);
+	*value_pending = (bit >= 0);
 
 	return res;
 }
 
-void notif_send_async(uint32_t value, uint16_t guest_id __maybe_unused)
+void notif_send_async(uint32_t value, uint16_t guest_id)
 {
+	struct guest_partition *prtn = NULL;
+	struct notif_vm_bitmap *nvb = NULL;
 	uint32_t old_itr_status = 0;
 	struct itr_chip *itr_chip = interrupt_get_main_chip();
 
-	assert(value <= NOTIF_ASYNC_VALUE_MAX && !guest_id);
+
```

Comment:
Please remove this duplicated blank line that checkpatch complains over, I must have added it by mistake.

Reply:
Noted.